source/include/jobs.h

```diff
@@ -742,6 +780,85 @@ JobsStatus_t Jobs_Update( char * buffer,
                           size_t * outLength );
 /* @[declare_jobs_update] */
 
+/**
+ * @brief Populate a message string for an UpdateJobExecution request.
+ *
+ * @param status Current status of the job
+ * @param expectedVersion The version that is expected
+ * @param expectedVersionLength The length of the expectedVersion
+ * @param buffer The buffer to be written to
+ * @param bufferSize the size of the buffer
+ *
+ * @return 0 if write to buffer fails
+ * @return messageLength if the write is successful
+ */
+size_t Jobs_UpdateMsg( JobCurrentStatus_t status,
+                       char * expectedVersion,
+                       size_t expectedVersionLength,
+                       char * buffer,
+                       size_t bufferSize );
+
+/**
+ * @brief Retrieves the job ID from a given message (if applicable)
+ *
+ * @param message [In] A JSON formatted message which
+ * @param messageLength [In] The length of the message
+ * @param jobId [Out] The job ID
+ * @return size_t The job ID length
+ */
+size_t Jobs_GetJobId( const char * message,
+                      size_t messageLength,
+                      char ** jobId );
+
+/**
+ * @brief Retrieves the job document from a given message (if applicable)
+ *
+ * @param message [In] A JSON formatted message which
+ * @param messageLength [In] The length of the message
+ * @param jobDoc [Out] The job document
+ * @return size_t The length of the job document
+ */
+size_t Jobs_GetJobDocument( const char * message,
+                            size_t messageLength,
+                            char ** jobDoc );
+
+/**
+ * @brief Checks if a message comes from the start-next/accepted reserved topic
+ *
+ * @param topic The topic to check against
+ * @param topicLength The expected topic length
+ * @param thingName The device's thingName as registered with AWS IoT.
+ * @param thingNameLength The length of the thingName.
+ * @return true If the topic is the start-next/accepted topic
+ * @return false If the topic is not the start-next/accepted topic
+ */
+bool Jobs_IsStartNextAccepted( const char * topic,
+                               const size_t topicLength,
+                               const char * thingName,
+                               const size_t thingNameLength );
+
+/**
+ * @brief Checks if a message comes from the update/accepted reserved topic
+ *
+ * @param topic The topic to check against
+ * @param topicLength The expected topic length
+ * @param jobId Corresponding Job ID which the update was accepted for
+ * @param jobIdLength The Job ID length
+ * @param thingName The device's thingName as registered with AWS IoT.
+ * @param thingNameLength The length of the thingName.
+ * @param expectedStatus The job update status reported by AWS IoT Jobs
+ * @return true If the topic is the update/\<expectedStatus\> topic
+ * @return false If the topic is not the update/\<expectedStatus\> topic
```

Comment:
Are these escape sequences for doxygen?

Reply:
Yes they are, without them doxygen gives an error since its treating it like a tag